tests/test_fsoe_master.py

```diff
@@ -942,3 +951,301 @@ def test_insert_in_best_position(self, sample_safe_dictionary):
+        ]
+        added_crc = 0
+        for idx in range(9):
+            crc_uid = f"FSOE_SLAVE_FRAME_ELEM_CRC{idx}"
+            if crc_uid in existing_crcs:
+                continue
+            safe_dict._registers[self.AXIS_1][crc_uid] = EthercatRegister(
+                idx=0x6760,
+                subidx=len(existing_crcs) + added_crc,
+                dtype=RegDtype.U16,
+                access=RegAccess.RO,
+                identifier=crc_uid,
+                pdo_access=RegCyclicType.SAFETY_INPUT,
+                cat_id="FSOE",
+            )
+            added_crc += 1
+        # Create safe dictionary
+        fsoe_dict = FSoEMasterHandler.create_safe_dictionary(safe_dict)
+
+        maps = PDUMaps.empty(fsoe_dict)
+
+        for idx in range(9):
+            test_uid = f"TEST_SI_U16_{idx}"
+            maps.inputs.add(fsoe_dict.name_map[test_uid])
+
+        validator = FSoEDictionaryMapValidator()
+        exceptions = validator.validate_dictionary_map_fsoe_frame_rules(
+            maps.inputs, rules=[FSoEFrameRules.SAFE_DATA_BLOCKS_VALID]
+        )
+        assert len(exceptions) == 1
+        assert FSoEFrameRules.SAFE_DATA_BLOCKS_VALID in exceptions
+        exception = exceptions[FSoEFrameRules.SAFE_DATA_BLOCKS_VALID]
+        assert isinstance(exception, InvalidFSoEFrameRule)
+        assert "Expected 1-8 safe data blocks, found 9" in exception.exception
+        assert exception.position is None
+        assert validator.is_rule_valid(FSoEFrameRules.SAFE_DATA_BLOCKS_VALID) is False
+
+    @pytest.mark.fsoe
+    def test_validate_safe_data_blocks_objects_split_across_blocks(self, sample_safe_dictionary):
+        """Test that SafeDataBlocksValidator fails when <= 16 bits objects are split."""
+        _, fsoe_dict = sample_safe_dictionary
+        maps = PDUMaps.empty(fsoe_dict)
+
+        maps.inputs.add(fsoe_dict.name_map[STOFunction.COMMAND_UID])
+        maps.inputs.add_padding(bits=6)
+        maps.inputs.add(fsoe_dict.name_map[SS1Function.COMMAND_UID.format(i=1)])
+        maps.inputs.add(fsoe_dict.name_map[SS2Function.COMMAND_UID.format(i=1)])
+        maps.inputs.add(fsoe_dict.name_map[self.TEST_SI_U8_UID])
+
+        # Test that rule fails because the 8-bit object is split across blocks
+        validator = FSoEDictionaryMapValidator()
+        exceptions = validator.validate_dictionary_map_fsoe_frame_rules(
+            maps.inputs, rules=[FSoEFrameRules.OBJECTS_SPLIT_RESTRICTED]
+        )
+        assert len(exceptions) == 1
+        assert FSoEFrameRules.OBJECTS_SPLIT_RESTRICTED in exceptions
+        exception = exceptions[FSoEFrameRules.OBJECTS_SPLIT_RESTRICTED]
+        assert isinstance(exception, InvalidFSoEFrameRule)
+        assert exception.exception == (
+            "Data slot 0 contains an object that is not "
+            "32 bits, it cannot be split across multiple safe data blocks"
+        )
+        assert exception.position == [0, 1, 7, 8, 9]  # All items in the first block
```

Comment:
Position is in bits?
Will this be practical for showing to the user of the process image editor? @lucia-teruel